recurly/recurly-client-python
# adding support for new coupon duration attributes

**Description**:
Add support for new coupon duration attributes (duration, temporal_unit, temporal_amount)

**Additional approvers**:
@bhelx

**Testing**:
- Verify that new coupons can be created with duration=forever/single_use/temporal, single_use=true, & applies_for_months=n
- Verify that when reading all of the created coupons the values for duration, temporal_unit, temporal_amount, single_use, & applies_for_months are all correct

```
c = recurly.Coupon(
  name = str(uuid.uuid4()), coupon_code = str(uuid.uuid4()), discount_in_cents = recurly.Money(200),
  duration = 'forever'
)
c.save()
c = recurly.Coupon.get(c.coupon_code)
print c.duration
print c.temporal_unit
print c.temporal_amount
print c.applies_for_months
print c.single_use

c = recurly.Coupon(
  name = str(uuid.uuid4()), coupon_code = str(uuid.uuid4()), discount_in_cents = recurly.Money(200),
  duration = 'single_use'
)
c.save()
c = recurly.Coupon.get(c.coupon_code)
print c.duration
print c.temporal_unit
print c.temporal_amount
print c.applies_for_months
print c.single_use

c = recurly.Coupon(
  name = str(uuid.uuid4()), coupon_code = str(uuid.uuid4()), discount_in_cents = recurly.Money(200),
  duration = 'temporal', temporal_unit = 'month', temporal_amount = 3,
)
c.save()
c = recurly.Coupon.get(c.coupon_code)
print c.duration
print c.temporal_unit
print c.temporal_amount
print c.applies_for_months
print c.single_use

c = recurly.Coupon(
  name = str(uuid.uuid4()), coupon_code = str(uuid.uuid4()), discount_in_cents = recurly.Money(200),
  applies_for_months = 3
)
c.save()
c = recurly.Coupon.get(c.coupon_code)
print c.duration
print c.temporal_unit
print c.temporal_amount
print c.applies_for_months
print c.single_use

c = recurly.Coupon(
  name = str(uuid.uuid4()), coupon_code = str(uuid.uuid4()), discount_in_cents = recurly.Money(200),
  single_use = True
)
c.save()
c = recurly.Coupon.get(c.coupon_code)
print c.duration
print c.temporal_unit
print c.temporal_amount
print c.applies_for_months
print c.single_use
```

**Deployment instructions**:
On hold until server side feature is deployed
